midealocal/security.py

```diff
@@ -212,7 +212,9 @@ class MideaAirSecurity(CloudSecurity):
     def __init__(self, login_key: str) -> None:
         super().__init__(login_key, None, None)
 
-    def sign(self, url: str, data: Any, random: str) -> str:
+    def sign(self, url: str, data: dict[str, Any] | str, random: str) -> str:
+        if isinstance(data, str):

```

Comment:
This should be outside of this PR

Reply:
SO you prefer a single PR with the check and after this typing improvement ?